diff -up lz4-1.10.0/build/cmake/CMakeLists.txt.omv~ lz4-1.10.0/build/cmake/CMakeLists.txt
--- lz4-1.10.0/build/cmake/CMakeLists.txt.omv~	2024-07-23 16:22:18.673459860 +0200
+++ lz4-1.10.0/build/cmake/CMakeLists.txt	2024-07-23 16:30:11.535031922 +0200
@@ -64,6 +64,7 @@ endif(NOT LZ4_BUNDLED_MODE AND NOT CPack
 include(CMakeDependentOption)
 CMAKE_DEPENDENT_OPTION(BUILD_SHARED_LIBS "Build shared libraries" ON "NOT LZ4_BUNDLED_MODE" OFF)
 CMAKE_DEPENDENT_OPTION(BUILD_STATIC_LIBS "Build static libraries" OFF "BUILD_SHARED_LIBS" ON)
+OPTION(BUILD_BENCHMARKS ON)
 
 if(NOT BUILD_SHARED_LIBS AND NOT BUILD_STATIC_LIBS)
   message(FATAL_ERROR "Both BUILD_SHARED_LIBS and BUILD_STATIC_LIBS have been disabled")
@@ -71,6 +72,7 @@ endif()
 
 set(LZ4_LIB_SOURCE_DIR "${LZ4_TOP_SOURCE_DIR}/lib")
 set(LZ4_PROG_SOURCE_DIR "${LZ4_TOP_SOURCE_DIR}/programs")
+set(LZ4_TESTS_SOURCE_DIR "${LZ4_TOP_SOURCE_DIR}/tests")
 
 include_directories("${LZ4_LIB_SOURCE_DIR}")
 
@@ -144,6 +146,21 @@ if (LZ4_BUILD_CLI)
   set_target_properties(lz4cli PROPERTIES OUTPUT_NAME lz4)
 endif()
 
+if(BUILD_BENCHMARKS)
+  include_directories("${LZ4_PROG_SOURCE_DIR}")
+  add_executable(fullbench ${LZ4_TESTS_SOURCE_DIR}/fullbench.c)
+  target_link_libraries(fullbench lz4)
+  # xxhash namespace
+  if(BUILD_SHARED_LIBS)
+    target_compile_definitions(fullbench PRIVATE
+      XXH_NAMESPACE=LZ4_)
+  endif()
+  if(BUILD_STATIC_LIBS)
+    target_compile_definitions(fullbench PRIVATE
+      XXH_NAMESPACE=LZ4_)
+  endif()
+endif()
+
 # Extra warning flags
 if(MSVC)
   set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} /W4")
